R/make-seurat.R

```diff
@@ -0,0 +1,122 @@
+#' Convert an SCE object to Seurat
+#'
+#' Converts A SingleCellExperiment (SCE) object to Seurat format. This is
+#' primarily a wrapper around Seurat::as.Seurat() with some additional steps to
+#' include ScPCA metadata and options for converting the feature index from
+#' Ensembl gene ids to gene symbols.
+#'
+#' If present, reduced dimensions from SCE objects will be included, renamed to
+#' match Seurat default naming.
+#'
+#'
+#' @param sce The input SCE object
+#' @param use_symbols Logical indicated whether the Seurat object use gene
+#'   symbols for indexing. Default is TRUE.
+#' @param reference If `use_symbols` is TRUE, the reference to use for gene
+#'   symbols. One of `sce`, `scpca`, `10x2020`, `10x2024`, where `sce` uses the
+#'   symbols stored in the `gene_symbol` column of the SCE object's row data,
+#'   and the others use standardized translation tables. See `ensembl_to_symbol`
+#'   for more information. Default is `sce`.
+#' @param dedup_method Method to handle duplicated gene symbols. If `unique`,
+#'   the gene symbols will be made unique following standard Seurat procedures.
+#'   If `sum`, the expression values for each gene symbols will be summed.
+#' @param seurat_assay_version Whether to create Seurat `v3` or `v5` assay
+#'   objects. Default is `v3`.
+#'
+#' @return a Seurat object
+#' @export
+#'
+sce_to_seurat <- function(
+    sce,
+    use_symbols = TRUE,
+    reference = c("sce", "scpca", "10x2020", "10x2024"),
+    dedup_method = c("unique", "sum"),
+    seurat_assay_version = c("v3", "v5")) {
+  reference <- match.arg(reference)
+  dedup_method <- match.arg(dedup_method)
+  seurat_assay_version <- match.arg(seurat_assay_version)
+
+  stopifnot(
+    "Package `Seurat` must be installed for SCE to Seurat conversion." =
+      requireNamespace("Seurat", quietly = TRUE),
+    "Package `SeuratObject` must be installed for SCE to Seurat conversion." =
+      requireNamespace("SeuratObject", quietly = TRUE),
```

Comment:
If both of these are missing, I might like to know right away. That said, I do appreciate that there's no way to know _which_ is missing in my suggestion which is indeed a drawback of knowing right away.

```suggestion
    "Packages `Seurat` and `SeuratObject` must be installed for SCE to Seurat conversion." =
      requireNamespace("Seurat", quietly = TRUE) & requireNamespace("SeuratObject", quietly = TRUE),
```

Reply:
This is actually kind of redundant, as `Seurat` requires `SeuratObject`. One would have to do some real work to only have the former installed, which is why I check for `Seurat` first.